sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ObjectsShouldNotBeDisposedMoreThanOnce.cs

```diff
@@ -62,10 +62,15 @@ private static void CheckForMultipleDispose(CSharpExplodedGraph explodedGraph, S
             var objectDisposedCheck = new ObjectDisposedPointerCheck(explodedGraph);
             explodedGraph.AddExplodedGraphCheck(objectDisposedCheck);
 
+            // Store the nodes that should be reported and ignore duplicate reports for the same node.
+            // This is needed because we generate two CFG blocks for the finally statements and even
+            // though the syntax nodes are the same, when there is a return inside a try/catch block
+            // the walked CFG paths could be different and FPs will appear.
+            var nodesToReport = new Dictionary<SyntaxNode, string>();

```

Comment:
False positive appeared here. We started generating two issues for the same syntax node because it is now present in two CFG blocks (each finally block generates two CFG blocks). We needed a way to deduplicate issues.